src/Microsoft.PowerShell.Commands.Utility/commands/utility/convert-HTML.cs

```diff
@@ -376,9 +426,33 @@ protected override void BeginProcessing()
 
             if (!_fragment)
             {
-                WriteObject("<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\"  \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">");
+                if(!Transitional)
+                {
+                    WriteObject("<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\"  \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">");
+                }
+                else
+                {
+                    WriteObject("<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\"  \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">");
+                }
                 WriteObject("<html xmlns=\"http://www.w3.org/1999/xhtml\">");
                 WriteObject("<head>");
+                if(_charsetSpecified)
+                {
+                    WriteObject("<meta charset=\"" + _charset + "\">");
+                }
+                if(_metaSpecified)
+                {
+                    foreach(string s in _meta.Keys)
+                    {
+                        if(s == "content-type" || s == "default-style" || s =="refresh")
+                        {
+                            WriteObject("<meta http-equiv=\"" + s + "\" content=\"" + _meta[s] + "\">");
+                        }
+                        else if (s == "application-name" || s == "author" || s == "description" || s == "generator" || s == "keywords" || s == "viewport") {
+                            WriteObject("<meta name=\"" + s + "\" content=\"" + _meta[s] + "\">");
+                        }
+                    }
```

Comment:
Rather than doing the validation above and going through the hashtable twice, I think it might be better to change these lines to a switch statement and have it default to an error